ledger/heavy/executor/jetkeeper.go

```diff
@@ -136,26 +166,58 @@ func (jk *dbJetKeeper) AddHotConfirmation(ctx context.Context, pn insolar.PulseN
 		return errors.Wrapf(err, "failed to save updated jets")
 	}
 
-	err := jk.updateTopSyncPulse(ctx, pn, id)
+	err := jk.updateTopSyncPulse(ctx, pn)
 	return errors.Wrapf(err, "AddHotConfirmation. propagateConsistency returns error")
 }
 
 func (jk *dbJetKeeper) AddDropConfirmation(ctx context.Context, pn insolar.PulseNumber, id insolar.JetID, split bool) error {
 	jk.Lock()
 	defer jk.Unlock()
 
-	inslogger.FromContext(ctx).Debug("AddDropConfirmation. pulse: ", pn, ". ID: ", id.DebugString())
+	inslogger.FromContext(ctx).Debug("AddDropConfirmation. pulse: ", pn, ". ID: ", id.DebugString(), ", Split: ", split)
 
 	if err := jk.updateDrop(ctx, pn, id, split); err != nil {
 		return errors.Wrapf(err, "AddDropConfirmation. failed to save updated jets")
 	}
 
-	err := jk.updateTopSyncPulse(ctx, pn, id)
+	err := jk.updateTopSyncPulse(ctx, pn)
+
+	return errors.Wrap(err, "propagateConsistency returns error")
+}
+
+func (jk *dbJetKeeper) AddBackupConfirmation(ctx context.Context, pn insolar.PulseNumber) error {
+	jk.Lock()
+	defer jk.Unlock()
+
+	inslogger.FromContext(ctx).Debug("AddBackupConfirmation. pulse: ", pn)
+
+	if err := jk.updateBackup(pn); err != nil {
+		return errors.Wrapf(err, "AddDropConfirmation. failed to save updated jets")
+	}
+
+	err := jk.updateTopSyncPulse(ctx, pn)
 
 	return errors.Wrap(err, "propagateConsistency returns error")
 }
 
-func (jk *dbJetKeeper) updateTopSyncPulse(ctx context.Context, pn insolar.PulseNumber, jetID insolar.JetID) error {
+func (jk *dbJetKeeper) updateBackup(pulse insolar.PulseNumber) error {
+	jets, err := jk.get(pulse)
+	if err != nil && err != store.ErrNotFound {
+		return errors.Wrapf(err, "updateBackup. can't get pulse: %d", pulse)
+	}
+
+	if len(jets) == 0 {
+		return errors.New("backup confirmation comes first. It's impossible")
```

Comment:
not very helpful error message

Reply:
It means that we don't expect that backup confirmation comes first. How to clarify error message?